### PR TITLE
add out variant of cudnn fused conv

### PR DESCRIPTION
**Pull request by makslevental**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61824
* #61821
* __->__ #61802
* #61801
* #61800

resubmitting because the last one was unrecoverable due to making changes incorrectly in the stack

Differential Revision: [D29812513](https://our.internmc.facebook.com/intern/diff/D29812513)

### CODE REVIEW

**Review by desertfire — APPROVED** [review]
LGTM

**Comment by makslevental** [comment]
@makslevental has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29812513).

File: aten/src/ATen/native/cudnn/ConvShared.cpp [hunk 1]
```diff
@@ -565,6 +555,23 @@ Tensor cudnn_convolution_add_relu(
...
+    const Tensor& z_t,
+    const c10::optional<Scalar>& alpha,
+    const c10::optional<Tensor>& bias_t,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    int64_t groups,
+    Tensor& output_t
+    ) {
   if (output_t.numel() == 0) {
```
Comment (ngimel) [inline aten/src/ATen/native/cudnn/ConvShared.cpp:575]: `cudnn_convolution_add_relu_out` is a public API, which means that users can send arbitrarily-sized `output_t` arguments to it. 0-sized `output_t` doesn't mean early exit, incorrectly-sized `output_t` has to either error out or be resized. We are trying to deprecate resizing of incorrectly sized `out` args, except resizing 0-element to correct size, so erroring out is better I'd say.

Comment (makslevental) [reply aten/src/ATen/native/cudnn/ConvShared.cpp:575]: @ngimel sounds good. suggestion for which error to use?

Comment (ngimel) [reply aten/src/ATen/native/cudnn/ConvShared.cpp:575]: You can use `at::native::resize_output` which will throw warning for now if output is of the wrong size, then at some point we'll switch it to error.

Comment (makslevental) [reply aten/src/ATen/native/cudnn/ConvShared.cpp]: @ngimel all set. if this is correct can you please approve.

**Review by ngimel — APPROVED** [review]

**Comment by makslevental** [comment]
@makslevental has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29812513).

**Comment by makslevental** [comment]
@makslevental has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29812513).

**Pull request closed without merging**